rspace-bench/src/test/scala/coop/rchain/rspace/bench/serialization/ModelSerializerBench.scala

```diff
@@ -0,0 +1,175 @@
+  @Fork(value = 1)
+  @Warmup(iterations = 5)
+  @Measurement(iterations = 10)
+  def kryoHeavyRoundTrip(bh: Blackhole, state: KryoModelBenchState) = {
+    val res = state.roundTripMany(state.heavyGnats)(state.serializer)
+    bh.consume(res)
+  }
+
+}
+
+abstract class ModelSerializerBenchState {
+  type TestGNAT = GNAT[Par, BindPattern, ListParWithRandom, TaggedContinuation]
+
+  implicit def serializer: Serialize2ByteBuffer[TestGNAT]
+
+  def roundTrip[A](in: A)(implicit s: Serialize2ByteBuffer[A]): A = {
+    val meta = s.encode(in)
+    val out  = s.decode(meta)
+    assert(out == in)
+    out
+  }
+
+  def roundTripMany[A](seq: Seq[A])(implicit s: Serialize2ByteBuffer[A]): Seq[A] =
+    seq.map(roundTrip(_))
+
+  import collection.immutable.Seq
+
+  import org.scalacheck._
+  import org.scalacheck.Arbitrary._
+
+  import coop.rchain.rspace.Serialize
+
+  implicit def arbitraryDatum[C, T](chan: C)(
+      implicit
+      arbT: Arbitrary[T],
+      serializeC: Serialize[C],
+      serializeT: Serialize[T]
+  ): Arbitrary[Datum[T]] =
+    Arbitrary(for {
+      t <- arbT.arbitrary
+      b <- Arbitrary.arbitrary[Boolean]
+    } yield Datum.create(chan, t, b))
+
+  def arbitraryWaitingContinuation[C, P, K](chans: List[C])(
+      implicit
+      arbP: Arbitrary[P],
+      arbK: Arbitrary[K],
+      serializeC: Serialize[C],
+      serializeP: Serialize[P],
+      serializeK: Serialize[K]
+  ): Arbitrary[WaitingContinuation[P, K]] =
+    Arbitrary(
+      for {
+        pats         <- Gen.containerOfN[List, P](chans.length, arbP.arbitrary)
+        continuation <- arbK.arbitrary
+        boolean      <- Arbitrary.arbitrary[Boolean]
+      } yield WaitingContinuation.create(chans, pats, continuation, boolean)
+    )
+
+  implicit def arbitraryGnat()(
+      implicit
+      serializeC: Serialize[Par],
+      serializeP: Serialize[BindPattern],
+      serializeA: Serialize[ListParWithRandom],
+      serializeK: Serialize[TaggedContinuation]
+  ): Arbitrary[TestGNAT] =
+    Arbitrary(Gen.sized { size =>
+      val constrainedSize = if (size > 1) size else 1
+      for {
+        chans <- Gen.containerOfN[List, Par](constrainedSize, Arbitrary.arbitrary[Par])
+        data <- Gen.nonEmptyContainerOf[List, Datum[ListParWithRandom]](
+                 arbitraryDatum[Par, ListParWithRandom](chans.head).arbitrary
+               )
+        wks <- Gen
+                .nonEmptyContainerOf[List, WaitingContinuation[BindPattern, TaggedContinuation]](
+                  arbitraryWaitingContinuation[Par, BindPattern, TaggedContinuation](chans).arbitrary
+                )
+      } yield GNAT(chans, data, wks)
+    })
+
+  import coop.rchain.rholang.interpreter.storage.implicits._
+
+  val initSeed = 123456789L
+
+  val gnat: TestGNAT = gnat(10)
+
+  def gnat(size: Int): TestGNAT =
+    arbitraryGnat.arbitrary.apply(Parameters.default.withSize(size), Seed(initSeed + size)).get
+
+  val heavyGnats = (1 to 10).map(i => gnat(i))
```

Comment:
I'd expect at least a 1000 different gnats, with separate benchmarks for different gnat sizes.

So e.g.
 - roundTrip {proto,kryo} 1000 gnats of size {1, 10, 100}

The 'single gnat' benchmark is not useful, I'd say. Let's drop it.

Reply:
Please also add variants where Par and Expr are used instead of GNAT. So:

roundTrip {proto,kryo} 1000 {GNAT, Par, Expr}s of size {1, 10, 100}

Reply:
OK, but wouldn't such benchmarks belong to `modelBenchmarks` project? Here the context is RSpace, and serialization of it's internal `GNAT` type